src/lib.rs

```diff
@@ -1,32 +1,37 @@
 #![deny(missing_docs)]
 #![cfg_attr(test, deny(warnings))]
-#![allow(clippy::match_wild_err_arm)]
 
 //! # Stdio-Override
 //!
 //! This crate provides a library for overriding Stdio file descriptors. <br>
-//! It provides a Guard for the replacement so that when the guard is dropped the file descriptors are switched back
+//! It provides a guard for the replacement so that when the guard is dropped the file descriptors are switched back
 //! and the replacement File Descriptor will be closed.
 //!
-//! **Trying to replace an std File Descriptor twice without dropping the guard will result in a panic** <br>
+//! You can replace a standard stream twice, just keep in mind that each guard, when dropped, will
+//! replace stdout with the stdout that existed when it was created. This means that if you don't
+//! drop the guards in reverse order of their creation you won't end up back where you started.
```

Comment:
We could get the best of both worlds by having an atomic number that counts the number of active overrides. Each override will increment the counter when it is created and store the old value, and if when it's dropped the counter value is different it will panic. I will implement this.